internal/pkg/expanders/graph.go

```diff
@@ -0,0 +1,595 @@
+package expanders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/awesome-gocui/gocui"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/editor"
+	"github.com/lawrencegripper/azbrowse/internal/pkg/interfaces"
+	"github.com/lawrencegripper/azbrowse/pkg/armclient"
+)
+
+// NewGraphExpander creates a new instance of GraphExpander
+func NewGraphExpander(armclient *armclient.Client, gui *gocui.Gui, commandPanel interfaces.CommandPanel, contentPanel interfaces.ItemWidget) *GraphExpander {
+	return &GraphExpander{
+		client:       &http.Client{},
+		armClient:    armclient,
+		gui:          gui,
+		commandPanel: commandPanel,
+		contentPanel: contentPanel,
+	}
+}
+
+// Check interface
+var _ Expander = &GraphExpander{}
+
+func (e *GraphExpander) setClient(c *armclient.Client) {
+	e.armClient = c
+}
+
+const (
+	namespace = "graph"
+
+	itemTypeMe      = "graph/me"
+	itemTypeAppList = "/applications"
+	itemTypeSpList  = "/serviceprincipals"
+	itemTypeAppOrSp = "graph/apporsp"
+
+	actionUpdateItem      = "updateitem"
+	actionSearch          = "search"
+	actionViewOwners      = "owners"
+	actionListMyApps      = "listmyapps"
+	actionListDeletedApps = "deletedapps"
+	actionGetAppByID      = "getappbyid"
+	actionNewApp          = "newapp"
+)
+
+// MeResponse used for the /me response
+type MeResponse struct {
+	Mail        string `json:"mail"`
+	DisplayName string `json:"displayName"`
+}
+
+// AppsResponse used for the Apps / Service Principal list
+type AppsResponse struct {
+	ID          string `json:"id"`
+	DisplayName string `json:"displayName"`
+}
+
+// AppsListResponse is a wrapper for AppsResponse
+type AppsListResponse struct {
+	Items []AppsResponse `json:"value"`
+}
+
+// GraphExpander expands the Graph resource
+type GraphExpander struct {
+	ExpanderBase
+	client       *http.Client
+	armClient    *armclient.Client
+	commandPanel interfaces.CommandPanel
+	contentPanel interfaces.ItemWidget
+	gui          *gocui.Gui
+}
+
+// Name returns the name of the expander
+func (e *GraphExpander) Name() string {
+	return "GraphExpander"
+}
+
+// DoesExpand checks if this node can/should expand
+func (e *GraphExpander) DoesExpand(ctx context.Context, currentItem *TreeNode) (bool, error) {
+	if currentItem.Namespace == namespace {
+		return true, nil
+	}
+	return false, nil
+}
+
+// Expand returns items from Graph
+func (e *GraphExpander) Expand(ctx context.Context, currentItem *TreeNode) ExpanderResult {
+
+	switch currentItem.ItemType {
+	case itemTypeMe:
+		return e.showMe(ctx, currentItem)
+	case itemTypeSpList:
+		return e.listAppsOrSps(ctx, currentItem, "")
+	case itemTypeAppList:
+		return e.listAppsOrSps(ctx, currentItem, "")
+	case itemTypeAppOrSp:
+		return e.showAppOrSp(ctx, currentItem)
+	case "graph":
+		return e.listRootMenu(ctx, currentItem)
+	case "action":
```

Comment:
Is this required? (Honestly a question, slightly rusty on how actions hook up)

My recollection is that `ListActions` and `ExecuteActions` are going to handle this and `Expand` only matters for treenodes.

Reply:
I have it there because I was previously seeing the root menu get re-displayed underneath search results :/